x/chainlink/keeper/keeper_test.go

```diff
@@ -294,13 +296,359 @@ func TestKeeper_GetLatestRoundId(t *testing.T) {
+			modify: param{submissionCount: 11, heartbeatTrigger: 22, deviationThresholdTrigger: 33, feedReward: &types.FeedRewardSchema{Amount: 44, Strategy: "abc"}},
+		},
+		{
+			feedId: "feed2",
+			insert: param{submissionCount: 100, heartbeatTrigger: 200, deviationThresholdTrigger: 300, feedReward: &types.FeedRewardSchema{Amount: 400, Strategy: "none"}},
+			modify: param{submissionCount: 101, heartbeatTrigger: 202, deviationThresholdTrigger: 303, feedReward: &types.FeedRewardSchema{Amount: 404, Strategy: "xyz"}},
+		},
+	}
+
+	for _, tc := range testCases {
+		testName := fmt.Sprintf("feed:%s", tc.feedId)
+		t.Run(testName, func(t *testing.T) {
+			k.SetFeed(ctx, &types.MsgFeed{
+				FeedId:                    tc.feedId,
+				SubmissionCount:           tc.insert.submissionCount,
+				HeartbeatTrigger:          tc.insert.heartbeatTrigger,
+				DeviationThresholdTrigger: tc.insert.deviationThresholdTrigger,
+				FeedReward:                tc.insert.feedReward,
+			})
+
+			_, _, err := k.SetSubmissionCount(ctx, &types.MsgSetSubmissionCount{
+				FeedId:          tc.feedId,
+				SubmissionCount: tc.modify.submissionCount,
+			})
+			require.NoError(t, err)
+
+			_, _, err = k.SetHeartbeatTrigger(ctx, &types.MsgSetHeartbeatTrigger{
+				FeedId:           tc.feedId,
+				HeartbeatTrigger: tc.modify.heartbeatTrigger,
+			})
+			require.NoError(t, err)
+
+			_, _, err = k.SetDeviationThresholdTrigger(ctx, &types.MsgSetDeviationThresholdTrigger{
+				FeedId:                    tc.feedId,
+				DeviationThresholdTrigger: tc.modify.deviationThresholdTrigger,
+			})
+			require.NoError(t, err)
+
+			_, _, err = k.SetFeedReward(ctx, &types.MsgSetFeedReward{
+				FeedId:     tc.feedId,
+				FeedReward: tc.modify.feedReward,
+			})
+			require.NoError(t, err)
+
+			result := k.GetFeed(ctx, tc.feedId)
+			require.Equal(t, tc.feedId, result.GetFeed().GetFeedId())
+			require.Equal(t, tc.modify.submissionCount, result.GetFeed().GetSubmissionCount())
+			require.Equal(t, tc.modify.heartbeatTrigger, result.GetFeed().GetHeartbeatTrigger())
+			require.Equal(t, tc.modify.deviationThresholdTrigger, result.GetFeed().GetDeviationThresholdTrigger())
+			require.Equal(t, tc.modify.feedReward, result.GetFeed().GetFeedReward())
+		})
+	}
+}
+
+func TestKeeper_DistributeReward(t *testing.T) {
 	t.Skip("TODO")
 }
+
+func TestKeeper_FeedOwnershipTransfer(t *testing.T) {
+	k, ctx := setupKeeper(t)
+
+	oldFeedOwner := GenerateAccount()
+	newFeedOwner := GenerateAccount()
+
+	feedToInsert := types.MsgFeed{
+		FeedId:    "feed1",
+		FeedOwner: oldFeedOwner,
+	}
+
+	// store feed with old owner
+	k.SetFeed(ctx, &feedToInsert)
+
+	// check if old owner is set
+	result := k.GetFeed(ctx, feedToInsert.GetFeedId())
+	require.Equal(t, feedToInsert.GetFeedId(), result.GetFeed().GetFeedId())
+	require.Equal(t, oldFeedOwner, result.GetFeed().GetFeedOwner())
+
+	// transfer ownership to new owner
+	_, _, err := k.FeedOwnershipTransfer(ctx, &types.MsgFeedOwnershipTransfer{
```

Comment:
should we also check for `signer?

Reply:
At the unit test level of keeper methods, `signer` is not used here, I just kept the used parameters in my tests

Reply:
ok, sounds good to me